American Whitewater/ViewControllers/Reaches/RunsListViewController.swift

```diff
@@ -374,24 +374,23 @@ class RunsListViewController: UIViewController {
     }
     
     @IBAction func runnableFilterChanged(_ runnableSwitch: UISwitch) {
-        DefaultsManager.runnableFilter = runnableSwitch.isOn
+        DefaultsManager.shared.runnableFilter = runnableSwitch.isOn
         fetchRiversFromCoreData()
     }
 
+    /// Returns the distance from the last saved user location to a location given as lat/long strings, in miles (approximately)
     private func calculateDistanceToRiver(riverLatString: String?, riverLonString: String?) -> Double? {
-        
         // get river info
         guard let riverLat = Double(riverLatString ?? ""), let riverLon = Double(riverLonString ?? "") else { return nil }
         let riverLocation = CLLocation(latitude: riverLat, longitude: riverLon)
         
         // get user location and check it
-        let currentLatitude = DefaultsManager.latitude
-        let currentLongitude = DefaultsManager.longitude
-        if currentLatitude > 1 && currentLongitude < 0 {
-            
-            let currentUserLocation = CLLocation(latitude: currentLatitude, longitude: currentLongitude)
-            
-            return currentUserLocation.distance(from: riverLocation) / 1609
+        let location = DefaultsManager.shared.location
+        
+        // FIXME: why is this check here? how does this check validity?
+        // Perhaps what is wanted instead is CLLocationCoordinate2DIsValid(), but I'm not sure what the intent is
```

Comment:
Yeah I think this checks northern and western hemisphere. Maybe that should be our general validity check that we use everywhere.

Reply:
We're missing a few bounds checks for that, right?

Also is that guaranteed to not be too strict (I thought I saw some international stuff in the region codes?)